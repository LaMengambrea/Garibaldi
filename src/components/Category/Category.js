import React, {Component} from 'react';

export default class Category extends Component {
  transformCategoryName(event) {
      console.log(event.target.innerHTML)
  }

  transformCategoryValue(event) {
      console.log(event.target.innerHTML)
  }

  render() {
    let classNamesForName = `CategoryName ${this.props.category.required} ${this.props.category.editableName}`;
    let classNamesForValue = `CategoryValue ${this.props.category.required} ${this.props.category.editableValue}`;
    return (
      <div className="row Category">
          <div className={classNamesForName} onClick={this.transformCategoryName}>{this.props.category.categoryName}</div>
          <div className={classNamesForValue} onClick={this.transformCategoryValue}>{this.props.category.categoryValue}</div>
      </div>
    );
  }
}