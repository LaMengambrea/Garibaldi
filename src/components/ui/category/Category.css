div.Category{
    padding-bottom: 5px;
}

.CategoryName {
    float: left;
    font-weight: bold;
}

.EditingCategory{
    border: 0px;
    background-color: transparent;
    border-bottom: 1px solid rgb(158, 158, 158);
}

.EditingCategory:focus{
    outline: none;
}

.CategoryInformation {
    float: left;
    width: 87%;
}

.CategoryActions {
    float: right;
}