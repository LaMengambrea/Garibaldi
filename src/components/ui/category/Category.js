import React, {Component} from 'react';
import PropTypes from 'prop-types';
import InlineEdit from 'react-edit-inline';
require('./Category.css')

export default class Category extends Component {
  customValidateText(text) {
      return true
  }

  dataChanged(data) {
      console.log(data)
  }

  render() {
    let classNamesForName = `CategoryName ${this.props.category.required}`;
    let classNamesForValue = `CategoryValue ${this.props.category.required}`;

    let categoryNameComponent = <InlineEdit
              className={classNamesForName}
              activeClassName="EditingCategory"
              text={this.props.category.categoryName}
              paramName="message"
              change={this.dataChanged}
            />

    let categoryValueComponent = <InlineEdit
              className={classNamesForValue}
              activeClassName="EditingCategory"
              text={this.props.category.categoryValue}
              paramName="message"
              change={this.dataChanged}
            />

    if(!this.props.category.editableName) {
      categoryNameComponent = <div className={classNamesForName}>{this.props.category.categoryName}</div>
    }

    if(!this.props.category.editableValue) {
      categoryValueComponent = <div className={classNamesForValue}>{this.props.category.categoryValue}</div>
    }

    return (
      <div className="row Category">
          {categoryNameComponent}: {categoryValueComponent} 
      </div>
    );
  }
}

Category.displayName = 'Category'

Category.propTypes = {
  category: PropTypes.object
};