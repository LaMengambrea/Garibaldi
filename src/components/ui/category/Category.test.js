import Category from './Category';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

let props = {
    category:{}
}

it('Category renders', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
        <Category {...props}/>
    </MuiThemeProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});