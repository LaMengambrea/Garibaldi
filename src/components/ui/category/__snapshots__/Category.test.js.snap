exports[`test Category renders 1`] = `
<div
  className="row Category">
  <div
    className="CategoryName undefined" />
  : 
  <div
    className="CategoryValue undefined" />
</div>
`;
